stdlib/LinearAlgebra/src/bidiag.jl

```diff
@@ -189,6 +189,10 @@ similar(B::Bidiagonal, ::Type{T}) where {T} = Bidiagonal(similar(B.dv, T), simil
 # The method below is moved to SparseArrays for now
 # similar(B::Bidiagonal, ::Type{T}, dims::Union{Dims{1},Dims{2}}) where {T} = spzeros(T, dims...)
 
+# Usually, reducedim_initarray calls similar, which yields a sparse matrix for a
+# Bidiagonal matrix. However, reducedim should yield a dense vector to increase performance.
+Base.reducedim_initarray(A::Bidiagonal, region, init) = fill(init, Base.reduced_indices(A,region))
```

Comment:
It seems suboptimal to repeat similar definitions and comments. How about doing this at a single location with a single `Union`?

Reply:
Okay, I'll do that. I've coded it in this way to have the appropriate methods for each type in the file for that type.